### PR TITLE
Nix flake

### LINKED ISSUE #50: Nix flake and NixOS module
Hey, if I opened a PR making this repo a flake, would you accept it?
I would also write a NixOS module to run Blockfrost.

Actually this is something that I'll do next days anyway, but obviously I would prefer keeping it upstream.
Just tell me if you want me to open a PR.

### PR DESCRIPTION
**Pull request by aciceri**
Closes #50 (partially, only the flake part, not the NixOS module) and #58

### CODE REVIEW

**Comment by aciceri** [comment]
@mmahut I'm implementing the flake wrapping the already present `default.nix` file, the only change I did (and that I'll ever do probably) to the `default.nix` file is making it use by default the same `nixpkgs` rev locked by the flake lockfile. Is this acceptable?

Right now I'm using a more recent `nixpkgs` that the one which was set by default in the `default.nix`, upgrading it implies that I had to add `name` (and change `machine` to `node.machine`) to `makeTest`s in the `default.nix` file.
If you don't like this I can downgrade the `nixpkgs` version used by the flake and at this point the `default.nix` file would emit exactly the same derivations as without my PR.

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      packages = forAllSystems (system: {
+        inherit (default.${system}) blockfrost-backend;
+        dockerImage = legacyPkgs.${system}.dockerTools.buildImage {
+          name = "blockfrost";
+          config = {
+            Cmd = [ "${self.packages.${system}.blockfrost-backend}/bin/blockfrost-backend" ];
+          };
+        };
+      });
+      # FIXME checks are not building
```
Comment (sorki) [inline flake.nix:27]: I think it's safe for you to exclude these. They require disabled sandboxing as they access external database.

Comment (aciceri) [reply flake.nix:27]: Do you prefer that I completely remove them or we can leave them somewhere in the flake? Pperhaps conditionally only if the `--impure` flag is passed (never did something like this but should be definitely possible)

**Comment by aciceri** [comment]
@sorki @1000101 @mmahut I also added a docker image generated with `dockerTools`, refer to #50 for the motivations. I'm keeping all changes in a single fork my simplicity.

**Comment by aciceri** [comment]
@sorki Let's do one thing at time i.e. let's address the NixOS module thing in another PR. Meanwhile this adds a flakes support and make it possible to build the Docker image with Nix.

**Comment by sorki** [comment]
> 



> @sorki Let's do one thing at time i.e. let's address the NixOS module thing in another PR. Meanwhile this adds a flakes support and make it possible to build the Docker image with Nix.

Sounds good and thanks for this one, testing.

**Comment by sorki** [comment]
Created a PR for your branch since I cannot seem to add commits here, PTAL https://github.com/mlabs-haskell/blockfrost-backend-ryo/pull/1

**Comment by aciceri** [comment]
@sorki Just resolved conflicts, it's ready to be merged I guess

**Comment by sorki** [comment]
> @sorki Just resolved conflicts, it's ready to be merged I guess

I've opened a rebased version at #83 so it runs thru CI, closing this one. Thanks!

**Comment by aciceri** [comment]
Thank you

**Pull request closed without merging**